### PR TITLE
Feature/mega menu alt

### PR DESCRIPTION
**Pull request by blake-hunsicker**
This PR contains the UI for the MegaMenu, but does not include search functionality or a working wallet connect. It is meant to be responsive and render well on all devices.

The MegaMenu file is in [/components/MegaMenu.tsx](https://github.com/MetaFam/TheGame/blob/feature/mega-menu-alt/packages/web/components/MegaMenu.tsx), and contains the UI for the new navbar.

The text and links for the nav bar are sourced from [utils/menuLinks.ts](https://github.com/MetaFam/TheGame/blob/feature/mega-menu-alt/packages/web/utils/menuLinks.ts).

### CODE REVIEW

**Comment by mquellhorst** [comment]
He @blake-hunsicker, nice work!

Some points:
The links should point to the following:
- Guilds -> my.metagame.wtf/guilds
- Guests -> my.metagame.wtf/quests

I would suggest placing a link to the balancer pool for planting seeds on the invest tab

I'll label the PR as WIP so you can finish it :)

**Comment by phys3** [comment]
> image
> The color of the menu items doesn't match the theme, and the height of the menu is too large

@hammadj i'll finish up this pr, do we have the design of the dropdowns for the mobile and desktop menu. i got access only for this https://www.figma.com/file/LAuXF0evRNBFv5djYlZUCi/Playbooks?node-id=1091%3A194960 which doesn't contain the expanded dropdowns

File: packages/web/utils/menuLinks.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      },
+    ],
+  },
+  {
+    label: 'invest',
+    menuItems: [
+      {
+        title: 'Seed',
+        explainerText:
+          'Find out all about the Seeds; why, where & how you should buy them',
```
Comment (dysbulic) [inline packages/web/utils/menuLinks.ts:137]:
```suggestion
          'Find out all about the SEEDs; why, where & how you should buy them',
```

I think the currency name is all caps, whether or not we're doing that here I don't know…

File: packages/web/utils/menuLinks.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      {
+        title: 'Grants',
+        explainerText:
+          'Find cool projects & people you could support financially',
+        url: 'https://wiki.metagame.wtf/',
+        icon: 'grants',
+      },
+      {
+        title: 'Balancer Pool',
+        explainerText: 'Plant your Seeds and watch them grow',
```
Comment (dysbulic) [inline packages/web/utils/menuLinks.ts:150]:
```suggestion
        explainerText: 'Plant your SEEDs and watch them grow',
```

File: packages/web/utils/menuLinks.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
+// in utils
+
+export interface MenuLink {
+  label: string;
+  icon?: any;
+  href?: string;
+  menuItems?: any;
```
Comment (dysbulic) [inline packages/web/utils/menuLinks.ts:7]:
```suggestion
```

File: packages/web/utils/menuLinks.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
+// in utils
+
+export interface MenuLink {
+  label: string;
+  icon?: any;
```
Comment (dysbulic) [inline packages/web/utils/menuLinks.ts:5]:
```suggestion
  icon?: string;
```

File: packages/web/utils/menuLinks.ts [hunk 5]
```diff
@@ -0,0 +1,157 @@
+// in utils
+
+export interface MenuLink {
```
Comment (dysbulic) [inline packages/web/utils/menuLinks.ts:3]: This interface is off. It nearly describes the nested objects, but is missing a couple fields. It is used as the type for the `MenuSectionLinks` which should just be something like:

```typescript
export interface MenuLinkSet {
  label: string;
  menuItems: MenuLink[];
}
```

File: packages/web/components/MegaMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,427 @@
...
+// Nav links on desktop -- text and links from utils/menuLinks.ts
+const DesktopNavLinks = () => (
+  <Flex
+    justifyContent="center"
+    alignContent="center"
+    fontFamily="exo"
+    display={{ base: 'none', lg: 'flex' }}
+    minW={{ base: 'auto', md: '40%' }}
+  >
+    {MenuSectionLinks.map((section: any) => (
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:153]:
```suggestion
    {MenuSectionLinks.map((section: MenuLinkSet) => (
```

File: packages/web/components/MegaMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,427 @@
...
+                )}
+              </MenuButton>
+              {section.menuItems.length > 3 ? (
+                <MenuList
+                  display="grid"
+                  gridTemplateColumns="repeat(2, 1fr)"
+                  width="948px"
+                  p="56px 6px 6px 56px"
+                >
+                  {section.menuItems.map((item: any) => (
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:185]:
```suggestion
                  {section.menuItems.map((item: MenuLink) => (
```

File: packages/web/components/MegaMenu.tsx [hunk 8]
```diff
@@ -0,0 +1,427 @@
...
+                    <DesktopMenuItem
+                      title={item.title}
+                      url={item.url}
+                      explainerText={item.explainerText}
+                      icon={item.icon}
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:190]:
```suggestion
                      {{ ...item }}
```

File: packages/web/components/MegaMenu.tsx [hunk 9]
```diff
@@ -0,0 +1,427 @@
...
+                  ))}
+                </MenuList>
+              ) : (
+                <MenuList
+                  display="grid"
+                  gridTemplateColumns="repeat(1, 1fr)"
+                  width="474px"
+                  p="56px 6px 6px 56px"
+                >
+                  {section.menuItems.map((item: any) => (
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:201]:
```suggestion
                  {section.menuItems.map((item: MenuLink) => (
```

File: packages/web/components/MegaMenu.tsx [hunk 10]
```diff
@@ -0,0 +1,427 @@
...
+                    <DesktopMenuItem
+                      title={item.title}
+                      url={item.url}
+                      explainerText={item.explainerText}
+                      icon={item.icon}
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:206]:
```suggestion
                      {{ ...item }}
```

File: packages/web/components/MegaMenu.tsx [hunk 11]
```diff
@@ -0,0 +1,427 @@
...
+    display={{ base: 'none', lg: 'flex' }}
+    justifyContent="flex-end"
+    // w='fit-content'
+    minW={{ base: '20%', lg: 'fit-content' }}
+    maxW={{ base: '20%', lg: 'fit-content' }}
+    p={2}
+    flex="1"
+    mt="auto"
+    mr="0"
+    mb="auto"
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:261]:
```suggestion
    my="auto"
    mr="0"
```

File: packages/web/components/MegaMenu.tsx [hunk 12]
```diff
@@ -0,0 +1,427 @@
...
+  >
+    <Badge
+      display={{ base: 'none', lg: 'flex' }}
+      flexDirection="row"
+      pt={2}
+      pr={4}
+      pb={2}
+      pl={4}
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:269]:
```suggestion
      px={4} py={2}
```

File: packages/web/components/MegaMenu.tsx [hunk 13]
```diff
@@ -0,0 +1,427 @@
...
+      bg="rgba(0,0,0,0.25)"
+      border="1px solid #2B2244"
+      borderRadius={50}
+      minW="fit-content"
+    >
+      <Image src={XPStar} alt="XP" height={14} width={14} />{' '}
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:275]:
```suggestion
      <Image src={XPStar} alt="XP" height={14} width={14} mr={3}/>
```

File: packages/web/components/MegaMenu.tsx [hunk 14]
```diff
@@ -0,0 +1,427 @@
...
+      </Text>
+    </Badge>
+    <Badge
+      display={{ base: 'none', lg: 'flex' }}
+      flexDirection="row"
+      m={2}
+      pt={2}
+      pr={4}
+      pb={2}
+      pl={4}
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:287]:
```suggestion
      px={4} py={2}
```

File: packages/web/components/MegaMenu.tsx [hunk 15]
```diff
@@ -0,0 +1,427 @@
...
+      bg="rgba(0,0,0,0.25)"
+      border="1px solid #2B2244"
+      borderRadius={50}
+      minW="fit-content"
+    >
+      <Image src={SeedMarket} alt="Seed" height={14} width={14} />{' '}
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:293]:
```suggestion
      <Image src={SeedMarket} alt="Seed" height={14} width={14} mr={3}/>
```

File: packages/web/components/MegaMenu.tsx [hunk 16]
```diff
@@ -0,0 +1,427 @@
...
+      <Text color="white" ml={[0, 0, 0, 2]}>
+        {pSeedBalance || 0}
+      </Text>
+    </Badge>
+    <Link href="profile/setup/username">
+      <Avatar
+        name="alt text"
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:300]:
```suggestion
        name={getPlayerName(player)}
```

File: packages/web/components/MegaMenu.tsx [hunk 17]
```diff
@@ -0,0 +1,427 @@
...
+        src={getPlayerImage(player)}
+        width="52px"
+        height="52px"
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:303]:
```suggestion
        w="52px" h="52px"
```

File: packages/web/components/MegaMenu.tsx [hunk 18]
```diff
@@ -0,0 +1,427 @@
...
+        justifyContent="space-between"
+        minH={{ base: '12vh', md: '10vh' }}
+        borderBottom="1px"
+        bg="rgba(0,0,0,0.5)"
+        borderColor="#2B2244"
+        sx={{ backdropFilter: 'blur(10px)' }}
+        pt={1.5}
+        pl={4}
+        pb={1.5}
+        pr={4}
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:332]:
```suggestion
        px={4} py={1.5}
```

File: packages/web/components/MegaMenu.tsx [hunk 19]
```diff
@@ -0,0 +1,427 @@
...
+        <Flex
+          onClick={menuToggle}
+          flexWrap="nowrap"
+          alignItems="center"
+          h="fit-content"
+          w="20%"
+          display={{ base: 'flex', lg: 'none' }}
+          p={2}
+          mt="auto"
+          mb="auto"
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:343]:
```suggestion
          my="auto"
```

File: packages/web/components/MegaMenu.tsx [hunk 20]
```diff
@@ -0,0 +1,427 @@
...
+        </Flex>
+        <Logo />
+        <DesktopNavLinks />
+        {/* <Search /> */}
+        {fetching ? (
+          <Spinner mt="18px" mr="24px" ml="162px" mb="26px" />
+        ) : (
+          <>
+            {isConnected && !!user?.player ? (
+              <PlayerStats player={user.player} pSeedBalance={pSeedBalance} />
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:359]:
```suggestion
              <PlayerStats player={user.player} {...{ pSeedBalance }} />
```

File: packages/web/components/MegaMenu.tsx [hunk 21]
```diff
@@ -0,0 +1,427 @@
...
+              <MetaButton
+                display={{ base: 'none', lg: 'block' }}
+                w="200px"
+                h="auto"
+                my="6px"
+                px="0"
+                ml="32px"
+                onClick={handleLoginClick}
+              >
+                Connect wallet
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:370]:
```suggestion
                Connect Wallet
```

File: packages/web/components/MegaMenu.tsx [hunk 22]
```diff
@@ -0,0 +1,427 @@
...
+        border="none"
+      >
+        {MenuSectionLinks.map((section) => (
+          <>
+            <Stack pt="16px">
+              <Text fontSize="16px" fontWeight="600" textTransform="capitalize">
+                {section.label}
+              </Text>
+              <SimpleGrid columns={2}>
+                {section.menuItems.map((item: any) => (
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:396]:
```suggestion
                {section.menuItems.map((item: MenuLink) => (
```

File: packages/web/components/MegaMenu.tsx [hunk 23]
```diff
@@ -0,0 +1,427 @@
...
+                  <Link
+                    display="flex"
+                    alignItems="center"
+                    fontSize="12px"
+                    href={item.url}
+                    border="1px"
+                    _odd={{ marginRight: '-1px' }}
+                    marginBottom="-1px"
+                    borderColor="purple.400"
+                    p="12px 16px"
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:406]:
```suggestion
                    px={4} py={3}
```

File: packages/web/components/MegaMenu.tsx [hunk 24]
```diff
@@ -0,0 +1,427 @@
...
+                  >
+                    <Avatar
+                      name="alt text"
+                      src={menuIcons[item.icon]}
+                      p="5px"
+                      width="24px"
+                      height="24px"
```
Comment (dysbulic) [inline packages/web/components/MegaMenu.tsx:413]:
```suggestion
                      w="24px" h="24px"
```

**Review by dysbulic — COMMENTED** [review]
Lots of little capitalization and redundancy things, but the basic logic looks solid.

I'd like to see fewer `px` values, but that's too big a pain to go back and fix.

File: packages/web/utils/menuLinks.ts [hunk 25]
```diff
@@ -0,0 +1,157 @@
...
+  label: string;
+  icon?: any;
+  href?: string;
+  menuItems?: any;
+}
+
+export const MenuSectionLinks: MenuLink[] = [
```
Comment (dysbulic) [inline packages/web/utils/menuLinks.ts:10]:
```suggestion
export const MenuSectionLinks: MenuLinkSet[] = [
```

**Comment by mquellhorst** [comment]
A couple of things about UX.
Originally the menu scrolls away with the content and on the players page the filters stick to the top of the page. \
Currently the menu sticks to the top of the page which results in the filters ending up behind the menu.
To solve this pls make the menu scroll up with the page

The Connect button is extremely large like this. I'ld prefer a regular button aligned in the center vertically.

**Review by mquellhorst — CHANGES_REQUESTED** [review]
A couple of things about UX.
Originally the menu scrolls away with the content and on the players page the filters stick to the top of the page.
Currently the menu sticks to the top of the page which results in the filters ending up behind the menu.
To solve this pls make the menu scroll up with the page

The Connect button is extremely large like this. I'ld prefer a regular button aligned in the center vertically.

**Comment by mquellhorst** [comment]
Thanks for the quick update!
I see a couple more things tho. Hopefully you can fix these then it looks mergable :)

1. On desktop it seems the menu is no longer in te center of the navbar.
2. On mobile the opacity of the background is lower than 1 making the links a bit hard to read.
3. The menu overlaps the Connect button on mobile.
4. On mobile you can scroll the menu but you can keep scrolling through the whole page.

Could you center it on desktop.
Make the opacity 1 on mobile and make it don't overlap the connect button and only being able to scroll the menu so the menu sticks

Hope this is clear enough else just dm me :)
Again, thanks for your work!

**Review by mquellhorst — CHANGES_REQUESTED** [review]
Thanks for the quick update!
I see a couple more things tho. Hopefully you can fix these then it looks mergable :)

    On desktop it seems the menu is no longer in te center of the navbar.
    On mobile the opacity of the background is lower than 1 making the links a bit hard to read.
    The menu overlaps the Connect button on mobile.
    On mobile you can scroll the menu but you can keep scrolling through the whole page.

Could you center it on desktop.
Make the opacity 1 on mobile and make it don't overlap the connect button and only being able to scroll the menu so the menu sticks

Hope this is clear enough else just dm me :)
Again, thanks for your work!

**Review by mquellhorst — APPROVED** [review]
Looks good, thanks!!

**Comment by mquellhorst** [comment]
Closes  #641

**Pull request merged**